tests/functional/oauth_webchannel_keys.js

```diff
@@ -0,0 +1,510 @@
+        '123done': true
+      });
+    },
+
+    'signup, verify same browser with original tab open': function () {
+      var self = this;
+
+      return openFxaFromRp(self, 'signup')
+        .execute(FunctionalHelpers.listenForWebChannelMessage)
+
+        .then(function () {
+          return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD, OLD_ENOUGH_YEAR);
+        })
+
+        .findByCssSelector('#fxa-confirm-header')
+        .end()
+
+        .then(function () {
+          return FunctionalHelpers.openVerificationLinkSameBrowser(
+                      self, email, 0);
+        })
+        .switchToWindow('newwindow')
+        // wait for the verified window in the new tab
+        .findById('fxa-sign-up-complete-header')
+        // the verification tab will almost always be the one to complete the flow.
+        // XXX TODO: can we not fail the test if the other tab wins?
+        // XXX TODO: or, ensure that the original tab always wins if it's open?
+        .then(testIsBrowserNotifiedOfLogin(self, { shouldCloseTab: false }))
+        .end()
+
+        .closeCurrentWindow()
+        // switch to the original window
+        .switchToWindow('')
+
+        .findById('fxa-sign-up-complete-header')
+        .end();
+    },
+
+    'signup, verify same browser with original tab closed': function () {
+      var self = this;
+
+      return openFxaFromRp(self, 'signup')
+
+        .then(function () {
+          return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD, OLD_ENOUGH_YEAR);
+        })
+
+        .findByCssSelector('#fxa-confirm-header')
+        .end()
+
+        .then(FunctionalHelpers.openExternalSite(self))
+
+        .then(function () {
+          return FunctionalHelpers.openVerificationLinkSameBrowser(self, email, 0);
+        })
+
+        .switchToWindow('newwindow')
+        .execute(FunctionalHelpers.listenForWebChannelMessage)
+
+        .then(testIsBrowserNotifiedOfLogin(self, { shouldCloseTab: false }))
+
+        .findById('fxa-sign-up-complete-header')
+        .end()
+
+        .closeCurrentWindow()
+        // switch to the original window
+        .switchToWindow('');
+    },
+
+    'signup, verify same browser, replace original tab': function () {
+      var self = this;
+
+      return openFxaFromRp(self, 'signup')
+
+        .then(function () {
+          return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD, OLD_ENOUGH_YEAR);
+        })
+
+        .findByCssSelector('#fxa-confirm-header')
+        .end()
+
+        .then(function () {
+          return FunctionalHelpers.getVerificationLink(email, 0);
+        })
+        .then(function (verificationLink) {
+          return self.get('remote').get(require.toUrl(verificationLink))
+            .execute(FunctionalHelpers.listenForWebChannelMessage);
+        })
+
+        .then(testIsBrowserNotifiedOfLogin(self, { shouldCloseTab: false }))
+
+        .findById('fxa-sign-up-complete-header')
+        .end();
+    },
```

Comment:
I'd like to leave it in unless you object stringly, since it might highlight weirdness when we're trying to do "complete in the original tab" in the future.